src/main/osd/osd_elements.c

```diff
@@ -778,6 +778,32 @@ static void osdElementCompassBar(osdElementParms_t *element)
     element->buff[9] = 0;
 }
 
+//display custom message from MSPv2
+static void osdElementCustomMsg1(osdElementParms_t *element)
+{
+    if (strlen(pilotConfig()->message[0]) == 0) {
+        strcpy(element->buff, "CUSTOM_MSG1");
+    } else {
+        strncpy(element->buff, pilotConfig()->message[0], strlen(pilotConfig()->message[0]));
+    }
+}
```

Comment:
```suggestion
static void osdElementCustomMsg(osdElementParms_t *element)
{
    unsigned msgIndex = element->item -  OSD_CUSTOM_MSG1;
    if (strlen(pilotConfig()->message[msgIndex]) == 0) {
        sprintf(element->buff, "CUSTOM_MSG%d", msgIndex + 1);
    } else {
        strncpy(element->buff, pilotConfig()->message[msgIndex], min(sizeof(element->buff, MAX_NAME_LENGTH));
    }
}
```

Reply:
Assign osdElementCustomMsg to all three elements. It will get called three times

Reply:
I see, thanks.